utils/src/shell_wrapper.rs

```diff
@@ -17,6 +20,13 @@ fn real_main() -> anyhow::Result<()> {
     // Therefore we dereference our symlink to get whatever it was originally.
     let shell = read_link(exe_dir.join("shell")).context("when locating the wrapped shell")?;
 
+    if shell.starts_with("/run/current-system/sw/bin/") {
+        while !Path::new("/run/current-system/sw/bin").exists() {
+            warn!("Activation script has not finished! Waiting for /run/current-system/sw/bin to exist");
```

Comment:
Can we do some inotify nonsense here? Or maybe write a oneshot pipe from the systemd wrapper?

Reply:
inotify is probably easier than the pipe. I'll look into it

Reply:
The example code in the [nix crate's docs](https://docs.rs/nix/latest/nix/sys/inotify/index.html) looks like we'd still need a loop to check for inotify events, so I don't know how much would be gained by that. The filesystem call every few seconds shouldn't put the system under any kind of additional load and the race condition happens really infrequently anyway, at least in user facing shells (i've seen it in kmsg a few times)

Reply:
You do need a loop, but it will actually sleep exactly until the event happens, instead of artificially delaying for 3 seconds.